src/pattern_matching/myers.rs

```diff
@@ -156,29 +446,374 @@ impl<'a, I: Iterator<Item = &'a u8>> Iterator for Matches<'a, I> {
     }
 }
 
+// 'Marker' traits
+pub struct Remember;
+pub struct NoRemember;
```

Comment:
Can you explain the purpose in the comment above? Would an enum make sense for the two values?

Reply:
'Traits' is completely wrong, I should have written 'structs'. The term 'remember' may not be the best word, in general the idea is that with `NoRemember`, only states needed for obtaining the alignment path at the *currently searched* position are stored. With `Remember`, states at each position in the text are stored. Consequently, the path can later be obtained for *any* position that was already searched (using `alignment_at` and related methods)